server/tst/FavoritesRouter/addFavorite.test.js

```diff
@@ -1,31 +1,53 @@
+import firestore from "../../src/config/firebase/firebaseConfig";
 import * as firebase from "firebase-admin";
-
 import addFavorite from "../../src/routers/FavoritesRouter/addFavorite";
-import firestore from "../../src/config/firebase/firebaseConfig";
+import firebaseTest from "../config/firebaseTest";
+import { E_BAD_FAVE_POST_REQ } from "../../src/config/constants";
 
 jest.mock("../../src/config/firebase/firebaseConfig");
-// jest.mock("firebase-admin");
 
 beforeEach(() => {
-    firebase.firestore.FieldValue = jest.fn();
-    firebase.firestore.FieldValue.arrayUnion = jest.fn();
-    firestore.doc = jest.fn();
-    firestore.doc.update = jest.fn();
+    firebaseTest();
     console.error = jest.fn();
+    addFavorite(123456789, 987654321, "DummyMenuItem");
+});
+
+test('addFavorite() -- basic normal function', async () => {
+    await expect(addFavorite(123456789, 987654321, "DummyMenuItem")).resolves.toBeUndefined();
+});
+
+test('addFavorite() -- favorites/menuItems doc', async () => {
+    await expect(firestore.doc).toHaveBeenCalledWith("favorites/menuItems");
```

Comment:
Why is this (and similar tests) its own test? Shouldn't the `toHaveBeenCalledWith` be something you call after a test of normal function?

Reply:
Updated